### PR TITLE
feat(select): add allowSelectAll option when isMulti is true

### PR DESCRIPTION
**Pull request by gregmartDOTin**
**Before submitting a pull request,** please make sure the following is done:

1. Fork [the repository](https://github.com/availity/availity-react) and create your branch from `master`.
2. Run `yarn` in the repository root.
3. If you've fixed a bug or added code that should be tested, add tests!
4. Ensure the test suite passes (`yarn test`). Tip: `yarn test --watch TestName` is helpful in development.
5. Make sure your code passed the conventional commits check. Read more about [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)

### CODE REVIEW

File: docs/source/form/select/components/select.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ Allow new items to be created if not found. **Default:** `false`.
 
 ### `allowSelectAll?: boolean`
 
-Adds a `Select all` option ( when `isMulti` is `true`).
+Adds a `Select all` option ( when `isMulti` is `true`). Note: allowSelectAll does not support use of the `loadOptions` prop.
```
Comment (nylon22) [inline docs/source/form/select/components/select.md]: nit: suggesting rephrase to:

Note: allowSelectAll is ignored when `loadOptions` is defined.

File: packages/select/src/Select.js [hunk 2]
```diff
@@ -45,6 +45,17 @@ const selectAllOption = {
   value: '*',
 };
 
+const validateSelectAllOptions = options => {
+  const filtered = options.filter(
+    option => option.value === selectAllOption.value
+  );
+  if (filtered.length > 0) {
+    console.warn(
```
Comment (nylon22) [inline packages/select/src/Select.js:54]: need to eslint disable this

File: packages/select/src/Select.js [hunk 3]
```diff
@@ -231,6 +243,12 @@ const Select = ({
     }
   }
 
+  if (attributes.loadOptions && allowSelectAll) {
+    console.warn(
```
Comment (nylon22) [inline packages/select/src/Select.js]: and here

**Review by nylon22 — CHANGES_REQUESTED** [review]

File: docs/source/form/select/components/select.md (hunk 1 above)
Comment (gregmartDOTin) [reply docs/source/form/select/components/select.md]: done

File: packages/select/src/Select.js (hunk 2 above)
Comment (gregmartDOTin) [reply packages/select/src/Select.js:54]: done

File: packages/select/src/Select.js (hunk 3 above)
Comment (gregmartDOTin) [reply packages/select/src/Select.js]: done

**Review by nylon22 — APPROVED** [review]
LGTM

**Pull request merged**